internal/tcpchecker/checkport.go

```diff
@@ -0,0 +1,66 @@
+package tcpchecker
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"time"
+)
+
+func CheckPort(address string, port int, reporter Reporter) (string, error) {
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%v", address, port), 1*time.Second)
+	if err != nil {
+		reporter.ReportPortCheck(Report{
+			Address: address,
+			Port:    port,
+			Open:    false,
+			Status:  "Connect error",
+			Error:   err,
+			Content: "",
+		})
+		return "", err
+	}
+
+	err = conn.SetDeadline(time.Now().Add(1 * time.Second))
+	if err != nil {
+		reporter.ReportPortCheck(Report{
+			Address: address,
+			Port:    port,
+			Open:    false,
+			Status:  "Set deadline error",
+			Error:   err,
+			Content: "",
+		})
+		return "", err
+	}
+
+	scanner := bufio.NewScanner(conn)
+
+	output := ""
+	for scanner.Scan() {
+		output += fmt.Sprintf("%s\n", scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		if netError, ok := err.(net.Error); ok && netError.Timeout() {
```

Comment:
To be a bit more resilient we can use Go 1.13's `errors.As()` instead. It will unwrap the error chain and assert instead of only doing so on the first.

```golang
var netError net.Error
if errors.As(err, &netError) && netError.Timeout() {
}
```

Reply:
Ok, will have a look at that 👍